File: balanced-brackets/C#/BalancedBrackets/BalancedBrackets.cs
namespace BalancedBracketsKata
{
    public class BalancedBrackets
    {
        public bool IsBalanced(string input)
        {
            Stack<char> stack = new Stack<char>();

            foreach (char c in input)
            {
                if (c == '[')
                {
                    stack.Push(c);
                }
                else if (c == ']')
                {
                    if (stack.Count == 0 || stack.Pop() != '[')
                    {
                        return false;
                    }
                }
            }

            return stack.Count == 0;
        }
    }
}

File: balanced-brackets/python/balanced_brackets.py
class BalancedBrackets:
    def is_balanced(self, input):
        stack = []

        for c in input:
            if c == '[':
                stack.append(c)
            elif c == ']':
                if not stack or stack.pop() != '[':
                    return False

        return len(stack) == 0
